typing/printtyp.ml

```diff
@@ -1495,7 +1520,7 @@ let unification_error unif tr txt1 ppf txt2 =
 
 let report_unification_error ppf env ?(unif=true)
     tr txt1 txt2 =
-  wrap_printing_env env (fun () -> unification_error unif tr txt1 ppf txt2)
+  wrap_printing_env env (fun () -> unification_error env unif tr txt1 ppf txt2)
```

Comment:
Note that the `env` received by `report_unification_error` was introduced to support `-short-paths`.
In some places, this is done in a best-effort way, falling back to the path itself when not available.
If you depend on it for what is defined, one would have to check that it is really accurate.
At least disable your check if `env` is `Env.empty`.